### PR TITLE
Update session object

### PR DESCRIPTION
**Pull request by fdemir**

### CODE REVIEW

**Review by aliosmandev — APPROVED** [review]

**Review by htkarabulut — COMMENTED** [review]
**Review by: OpenAI GPT-4
AI wants to: "Comment"**

1. Overall, the changes made in this PR look good. The addition of the new session object type in `next-auth.d.ts` is clear and the implementation of the JWT strategy for storing the user's username in the session object in `src/pages/api/auth/[...nextauth].tsx` is well-written.

2. I noticed that in the `jwt` callback function, there's a `TODO` comment. Make sure to address the comment if you plan to support more providers in the future. You could consider implementing a switch statement or a mapping function to handle different providers.

3. In the `jwt` callback function, you're using `token["profile"]` to assign a new value to the `profile` property of the token. While this will work, it's generally better to use the dot notation (`token.profile`) for property access. Additionally, the `@ts-ignore` comment can be removed if you ensure that the correct types are used.

4. The code has a clear structure and is easy to understand. The added comments and the use of the JWT strategy make it easy to follow and maintain. However, it would be helpful to include some additional comments in the `jwt` and `session` callback functions to explain their purpose for future maintainers.

Overall, the changes in this Pull Request are well done and enhance the 
functionality of the session object by adding the user's username.

---

Based on the review, I think the Pull Request can be approved after addressing the following minor changes:

1. Use the dot notation (`token.profile`) instead of the bracket notation (`token["profile"]`) for property access in the `jwt` callback function.

2. Remove the `@ts-ignore` comment and ensure the correct types are used.

3. Add some additional comments in the `jwt` and `session` callback functions to explain their purpose for future maintainers.

These changes are relatively minor, and once they are addressed, the Pull Request should be ready for approval.

**Pull request merged**
